docs/schemapack_spec.md

```diff
@@ -0,0 +1,164 @@
+# Schemapack Specification
+# =========================
+
+This document describes the structure and components of a `schemapack.yaml` file, which defines the schema used to validate and interpret datapacks. Each schemapack includes:
+
+* A schemapack version identifier
+
+* An optional schema description
+
+* A classes section, where each class represents a distinct entity type with associated content and its relations. Classes define the internal structure (content) of their instances using JSON Schema and may specify relationships (relations) to other classes defining constraints on how instances of different classes are linked, including their multiplicity and participation.
+
+### Keywords:
+
+`schemapack`: Specifies the Schemapack version the schema is built under.
+
+`description` *(optional)*: A short, human-readable summary of the schema.
+
+`classes`: A mapping of class names to *class definitions*. Class names must use PascalCase.
+
+Example:
+```yaml
+
+schemapack: 3.0.0
+description: Schema for experimental metadata
+classes:
+    Experiment:
+        ...
+```
+
+#### Class Definition
+
+A class definition includes the following properties:
+
+`description` *(optional)*: A brief explanation of the class.
+
+`id`: Defines the class's identifier property. Must follow this structure:
+
+* `propertyName`: The name of the identifier property. Must not conflict with names used in `content` or `relations`.
+
+* `description` *(optional)*: A human-readable description of the identifier.
+
+    Example:
+    ```yaml
+
+    id:
+        propertyName: alias
+        description: alias is id
+    ```
+
+`content`: A schema describing the structure of the class content. Must be a valid JSON Schema object. You may embed it inline or reference an external JSON or YAML file, which will be automatically loaded.
+
+```yaml
+
+content: content_schemas/Experiment.schema.json # <- path to the json schema
+```
+
+``` yaml
+content: # <- content schema embedded here
+    type: object
+    properties:
+        name:
+            type: string
+        description:
+            type: string
+```
+
+
+Given that the content of the Experiment.schema.json file is:
+```json
+{
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "description": {
+            "type": "string"
+        }
+    }
+}
+```
+
+`relations` *(optional)*: A mapping of relation names to *relation definitions*. Relation names must:
+  * Use `snake_case`.
+  * Start with a letter.
+  * Contain only alphanumeric characters and underscores.
```

Comment:
This also has the same uniqueness requirement as `propertyName`, right?

Reply:
Actually, no. It is evaluated independently; it does not have to be unique.

Reply:
What about https://github.com/ghga-de/schemapack/blob/9ff62c402067c0a752a8adb7264d87d5369bbd79/src/schemapack/_internals/spec/schemapack.py#L298C1-L314C1?